python/tchannel/tornado/tchannel.py

```diff
@@ -159,13 +172,57 @@ def handle_stream(self, stream, address):
         })
 
         log.debug(
-            "Successfully completed handshake with %s (%s)",
+            "Successfully completed handshake with %s:%s (%s)",
             conn.remote_host,
+            conn.remote_host_port,
             conn.remote_process_name,
         )
 
-        self.tchannel.peers.get(conn.remote_host).register_incoming(conn)
+        self.tchannel.peers.get(
+            "%s:%s" % (conn.remote_host,
+                       conn.remote_host_port)
```

Comment:
remote_host is only the ip. 
remote_host_port is for port. I changed it in the connection.py